test/stools_tests.jl

```diff
@@ -35,7 +35,7 @@ cpath = _setup_repo(false) # clean
 @test !endswith(gitdescribe(cpath), "-dirty")
 
 # tag!
-function _test_tag!(d, path, haspatch, DRWATSON_STOREPATCH)
+function _test_tag!(d, path, haspatch, has_message, DRWATSON_STOREPATCH)
```

Comment:
Its great that you modified this function to test for the existence of a message. However I see two problems: 1) you never call this function with `true` as the 4th argument, and hence, the function never actually tests the new code. 2) the function does not test whether the message itself is correct. Only tests that it is not the empty string.

I propose for you to _revert_ the modification of the `test_tag!` function. Instead, in the codeblock `@testset "message_name" begin` that you have created below, write a custom, hand coded test that explicitly calls `tag!` and then it explicitly tests that the tagged dictionary has exact information of the commit message, whatever that message is.

Reply:
I see, thank you! I will try to implement it